### PR TITLE
Use `to_regtypemod` on Postgres 17

### PR DESCRIPTION
**Pull request by theory**
When running on Postgres 17 or higher, take advantage of the new `to_regtypemod` function to much more correctly and efficiently normalize type names, including their type modifiers. Notably, interval types will now also be normalized, so note this fact in the docs.

Update the CI testing to test on Postgres 17 and adjust patch offsets for the line count change in `pgtap.sql.in`.

Finally a fully correct fix for #315 (cc @ewie)!

### CODE REVIEW

File: doc/pgtap.mmd [hunk 1]
```diff
@@ -4524,8 +4524,8 @@ pass the type as either "varchar(64)" or "character varying(64)". Example:
 
     SELECT col_type_is( 'myschema', 'sometable', 'somecolumn', 'timespantz(3)' );
```
Comment (ewie) [inline doc/pgtap.mmd:4525]: Not part of the patch, but would you please fix the typo in `timespantz`.

But `timestamp(3)` doesn't even match the paragraph above which talks about `varchar(64)`. Maybe it's used here because `timestamptz`/`timestamp with time zone` is similar to `varchar`/`character varying`.

**Review by ewie — CHANGES_REQUESTED** [review]

File: doc/pgtap.mmd (hunk 1 above)
Comment (theory) [reply doc/pgtap.mmd]: Done in b0184e4.

**Review by ewie — APPROVED** [review]
LGTM. Thanks!
